aiohttp_sentry/__init__.py

```diff
@@ -1,22 +1,49 @@
+import asyncio
 from functools import partial
+import sys
 
 import raven
 import raven_aiohttp
 
 
 class SentryMiddleware:
 
-    def __init__(self, sentry_kwargs=None):
+    def __init__(self, install_excepthook=True, sentry_kwargs=None):
         if sentry_kwargs is None:
             sentry_kwargs = {}
 
+        if install_excepthook:
+            # do not let raven.Client install its own excepthook
+            sentry_kwargs['install_sys_hook'] = False
+
         sentry_kwargs = {
             'transport': raven_aiohttp.AioHttpTransport,
             'enable_breadcrumbs': False,
             **sentry_kwargs,
         }
         self.client = raven.Client(**sentry_kwargs)
 
+        if install_excepthook:
+            self.update_excepthook()
+
+    def update_excepthook(self):
+        """Update sys.excepthook so that it closes the sentry transport."""
+
+        loop = asyncio.get_event_loop()
+        original_excepthook = sys.excepthook
+        def aiohttp_transport_excepthook(*exc_info):
+            """An aiohttp-transport-friendly excepthook.
+
+            It closes the transport, which delivers the messages."""
+            self.client.captureException(exc_info=exc_info, level='fatal')
+            transport = self.client.remote.get_transport()
+            if isinstance(transport, raven_aiohttp.AioHttpTransportBase):
+                # wait for Sentry transport to send the outstanding messages
+                loop.run_until_complete(transport.close())
+            original_excepthook(*exc_info)
```

Comment:
This part is pretty impressive, nice!

Reply:
The idea of using the original excepthook is taken from `raven.Client`:
https://github.com/getsentry/raven-python/blob/f6d79c3bcc25e804b6259fa9c4a6e030f9033bb2/raven/base.py#L280

Reply:
Well, it's a nice find either way 🙃 

Could you add a link to the original code in a comment?